pkg/trace/event/extractor_single_span_rate.go

```diff
@@ -0,0 +1,39 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package event
+
+import (
+	"github.com/DataDog/datadog-agent/pkg/remoteconfig/client/products/apmsampling"
+	"github.com/DataDog/datadog-agent/pkg/trace/pb"
+	"github.com/DataDog/datadog-agent/pkg/trace/sampler"
+)
+
+// singleSpanRateExtractor extracts spans that have been sampled using the single span sampling mechanism.
+type singleSpanRateExtractor struct{}
+
+func NewSingleSpanExtractor() Extractor {
+	return &singleSpanRateExtractor{}
+}
+
+func (e *singleSpanRateExtractor) Extract(s *pb.Span, _ sampler.SamplingPriority) (float64, bool) {
+	if len(s.Metrics) == 0 {
+		// metric not set
+		return 0, false
+	}
+	m, ok := s.Metrics[sampler.KeySpanSamplingMechanism]
```

Comment:
Maybe it's better to use `traceutil.GetMetric` just to make sure no `nil` surprises happen.

Reply:
I don't mind but we do check above if metrics isn't nil